src/hotspot/share/opto/idealGraphPrinter.cpp

```diff
@@ -862,24 +904,88 @@ void IdealGraphPrinter::walk_nodes(Node* start, bool edges) {
   }
 }
 
-void IdealGraphPrinter::print_graph(const char* name) {
+static bool skip_frame(const char* name) {
+  return strstr(name, "IdealGraphPrinter") != nullptr ||
+         strstr(name, "Compile::print_method") != nullptr ||
+         strstr(name, "Compile::igv_print_graph") != nullptr ||
+         strstr(name, "PrintBFS") != nullptr ||
+         strstr(name, "Node::dump_bfs") != nullptr;
+}
+
+static bool stop_frame_walk(const char* name) {
+  return strstr(name, "C2Compiler::compile_method") != nullptr;
+}
```

Comment:
Nit: You could write `must_stop_frame_walk`. Btw, it could be nice to have a comment to explain why this is the condition to stop on. Maybe that comment could then turn into an even better method name?

Reply:
Thanks, I went with `should_end_stack_walk`, and added a comment as requested (commit 1210180b).